source/router.ts

```diff
@@ -263,7 +268,14 @@ export class Router extends EventTarget {
 			constructedRoute.clientRoute.parent = parent && parent.clientRoute;
 			constructedRoute.clientRoute.component = constructedRoute.component;
 
-			this.constructedRoutes.push(constructedRoute);
+			// replace root route with default child route
```

Comment:
readable form the code, add more description why we do this

Reply:
// a default route should be resolved instead of the parent route whenever the path is requested
// replace the parent route with the default as a navigateable destination to automatically lead the router to the default route

Reply:
Done!